src/scenic/domains/driving/model.scenic

```diff
@@ -358,11 +358,11 @@ def withinDistanceToObjsInLane(vehicle, thresholdDistance):
     for obj in objects:
         if not (vehicle can see obj):
             continue
-        if not (network.laneAt(vehicle) == network.laneAt(obj) or network.intersectionAt(vehicle)==network.intersectionAt(obj)):
-            continue
         if (distance from vehicle.position to obj.position) < 0.1:
             # this means obj==vehicle
-            pass
-        elif (distance from vehicle.position to obj.position) < thresholdDistance:
-            return True
+            continue
+        if (network.intersectionAt(vehicle) is not None and network.intersectionAt(vehicle)==network.intersectionAt(obj)) or \
+            network.intersectionAt(vehicle) is None and network.laneAt(vehicle) == network.laneAt(obj):
+            if (distance from vehicle.position to obj.position) < thresholdDistance:
+                return True
```

Comment:
I personally find this harder to read than the old linear style. How about the following:
```
inter = network.intersectionAt(vehicle)
if inter and inter != network.intersectionAt(obj):    # different intersections
    continue
if not inter and network.laneAt(vehicle) != network.laneAt(obj):    # different lanes
    continue
if (distance from vehicle to obj) < thresholdDistance:
    return True
```

Reply:
Agreed, done

Reply:
Done